app/models/agents/usabilla_agent.rb

```diff
@@ -218,18 +218,33 @@ def retrieve_emails
 
     def usabilla_response_to_event(r)
       {
-        comment: r.comment,
-        score: r.rating,
+        comment: extract_comment(r),
+        score: extract_score(r),
         location: r.location,
         id: r.id,
         custom: r.custom,
-        public_url: r.public_url,
-        button_id: r.button_id,
+        public_url: extract_if_present(:public_url, r),
+        button_id: extract_if_present(:button_id, r),
         created_at: r.date,
-        email: r.email
+        email: extract_if_present(:email, r),
+        data: extract_if_present(:data, r)
```

Comment:
I think there is built in serialization, You can see it in the uploaded screenshot. But let me test with some event propagation.